code/game/turfs/floors/_floor.dm

```diff
@@ -12,98 +12,101 @@
 	zone_membership_candidate = TRUE
 	open_turf_type = /turf/open/airless
 
+	// Reagent to use to fill the turf.
+	var/fill_reagent_type
 	var/can_engrave = TRUE
 
 	// Damage to flooring.
 	// These are icon state suffixes, NOT booleans!
 	var/_floor_broken
 	var/_floor_burned
 
-	// Plating data.
-	var/base_name = "plating"
-	var/base_desc = "The naked hull."
-	var/base_icon = 'icons/turf/flooring/plating.dmi'
-	var/base_icon_state = "plating"
-	var/base_color = COLOR_WHITE
-
 	// Flooring data.
 	var/flooring_override
-	var/initial_flooring
+	var/decl/flooring/base_flooring = /decl/flooring/plating
 	var/decl/flooring/flooring
 
-// Defining this here as a dummy mapping shorthand so mappers can search for 'plating'.
-/turf/floor/plating
+	var/const/TRENCH_DEPTH_PER_ACTION = 100
 
-/turf/floor/can_climb_from_below(var/mob/climber)
-	return TRUE
+/turf/floor/Initialize(var/ml, var/floortype)
 
-/turf/floor/is_plating()
-	return !flooring
+	if(base_flooring)
+		base_flooring = GET_DECL(base_flooring)
 
-/turf/floor/get_base_movement_delay(var/travel_dir, var/mob/mover)
-	return flooring?.get_movement_delay(travel_dir, mover) || ..()
+	. = ..(ml)
 
-/turf/floor/protects_atom(var/atom/A)
-	return (A.level <= LEVEL_BELOW_PLATING && !is_plating()) || ..()
+	if(ispath(material))
+		if(ml)
+			set_turf_materials(material, skip_update = TRUE)
+			queue_icon_update()
+		else
+			set_turf_materials(material)
+	else if(!istype(material))
+		material = null
+		update_icon()
+
+	if(!floortype && ispath(flooring))
+		floortype = flooring
 
-/turf/floor/Initialize(var/ml, var/floortype)
-	. = ..(ml)
-	if(!floortype && initial_flooring)
-		floortype = initial_flooring
 	if(floortype)
 		if(ml)
 			set_flooring(GET_DECL(floortype), skip_update = TRUE)
 			queue_icon_update()
 		else
 			set_flooring(GET_DECL(floortype))
 
-/turf/floor/proc/set_flooring(var/decl/flooring/newflooring, skip_update)
+	if(fill_reagent_type && get_physical_height() < 0)
+		add_to_reagents(fill_reagent_type, abs(height))
```

Comment:
move the icon update code here and readd the neighbor update code, or else neighboring edges won't be added or removed on turf change
```suggestion
		add_to_reagents(fill_reagent_type, abs(height))
		
	if(material || flooring || base_flooring)
		if(ml)
			queue_icon_update()
		else
			for(var/direction in global.alldirs)
				var/turf/target_turf = get_step_resolving_mimic(src, direction)
				if(istype(target_turf))
					if(TICK_CHECK) // not CHECK_TICK -- only queue if the server is overloaded
						target_turf.queue_icon_update()
					else
						target_turf.update_icon()
			update_icon()
```